### ISSUE TITLE
Kraken achieves +97% Arabic

### ISSUE THREAD

**Issue by christophered** [tesseract-ocr/tesseract#892]
After doing much search I have found Kraken which is an open-source fork of ocropus that solves a number of issues including:

- clstm compatibility
- Right-to-left/BiDi support
- Explicit input/output handling
- Word and character bounding boxes in hOCR
- Removal of runtime dependency on gcc
- Clean public API
- ets ....

Stating Kraken's most important advantage, its workflow allows one to train new models relatively easily.

[In this research, they have managed to get 97% to 99% recognition rate of Arabic language, this rate excludes spacing and punctuation errors (2-5%).](https://drive.google.com/file/d/0BzDVkBcqiyEsbC16ZGktOWNiUDg/view)

What is really bothering me is that Kraken is a fork of ocropus that is using clstm, while Tesseract 4.x is the more complex and most sophisticated software that have improved ocropus techniques from scratch, but still can't achieve such practical results of Kraken.

Karken is the key to recognizing Arabic, Hebrew, Urdu, Persian/Farsi, etc....

@theraysmith @Shreeshrii @amitdo @zdenop 
I hope that we will be able to solve the Arabic Language in Tesseract 4.x similarly to Kraken.
Please have a look at the Kraken project, and their method as guidance:
https://github.com/mittagessen/kraken
http://kraken.re
https://drive.google.com/file/d/0BzDVkBcqiyEsbC16ZGktOWNiUDg/view

**Comment by bmwmy** [comment]
After testing kraken v0.9.1 using provided Arabic models, I believe that 97% is in theory even with the same font that used in the training 

same basic tests are attached
[Kraken_test_results.tar.gz](https://github.com/tesseract-ocr/tesseract/files/982008/Kraken_test_results.tar.gz)

**Comment by christophered** [comment]
@bmwmy have you tested training yourself?
Kraken developer told me that those models that you presented, he made, and are not that sophisticated, maybe just for testing.
Please try to train clstm, then reply back

**Comment by christophered** [comment]
@bmwmy have you trained using [train.sh](https://raw.githubusercontent.com/mittagessen/kraken-vagrant/master/training/train.sh)

**Comment by bmwmy** [comment]
No, actually I used the downloadable models which is on the repository.
Will try to train and will give feedback.

**Comment by christophered** [comment]
@bmwmy have you seen [this update by Ray](https://github.com/tesseract-ocr/langdata/commit/3ab6581a11eea90d4dc2ba46a811447ef231b644#diff-b8c77a8bc89ffb301d80233f2874c9d3),did he add the Arabic diacritics "Harakat"?
is that a good thing or a bad thing for ocr, what do you think?

**Comment by bmwmy** [comment]
Hi I used @christophered trained Arabic kraken model for about 75000 lines and did comparison with latest ara.traineddata with lstm. I used standard Arabic font e.g. (Times New Roman)
The result is that tesseract make it 100% except the bug of flipping combined letters but letters are correct (https://github.com/tesseract-ocr/tesseract/issues/758).
kraken mix up some letters with similar letters achieves ~97%.

for Arabic diacritics I think OCR engine should be aware of that but neglect it in the output or during ocr process.
